src/shared/install.c

```diff
@@ -3771,7 +3800,12 @@ int unit_file_get_list(
                             !strv_contains(states, unit_file_state_to_string(f->state)))
                                 continue;
 
-                        r = hashmap_put(h, basename(f->path), f);
+                        _cleanup_free_ char *fname = NULL;
+                        r = path_extract_filename(f->path, &fname);
+                        if (r < 0)
+                                return r;
+
+                        r = hashmap_put(h, fname, f);
```

Comment:
IIUC the CI errors are caused by this line - you put a pointer to the hashmap, but once `fname` here goes out of scope the memory for that pointer gets freed and subsequent `hashmap_get()` calls will return an invalid pointer.

Using something like:

```suggestion
                        r = hashmap_put(h, TAKE_PTR(fname), f);
```

should fix that.